File: apps/frontend/src/api/llmConfig.ts
export interface LlmConfig {
  model: string;
  base_url: string;
  api_key_set: boolean;
}

export async function getLlmConfig(): Promise<LlmConfig> {
  const res = await fetch('/api/llm_config');
  if (!res.ok) throw new Error(`Failed to fetch LLM config: ${res.status}`);
  return res.json();
}

export async function postLlmConfig(patch: {
  model?: string;
  base_url?: string;
  api_key?: string;
}): Promise<void> {
  const res = await fetch('/api/llm_config', {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(patch),
  });
  if (!res.ok) throw new Error(`Failed to update LLM config: ${res.status}`);
}

File: apps/backend/app/routes/llm_config.py
from fastapi import APIRouter

from app.core.llm import get_llm_config, set_llm_config
from app.models.llm import LLMConfig

router = APIRouter()


@router.get("/api/llm_config")
def get_llm_config_endpoint():
    """Return the active LLM config (API key masked)."""
    cfg = get_llm_config()
    return {
        "base_url": cfg["base_url"],
        "model": cfg["model"],
        "api_key_set": bool(cfg["api_key"]),
    }


@router.post("/api/llm_config")
def set_llm_config_endpoint(cfg: LLMConfig):
    """Hot-swap the LLM provider without restarting the server."""
    set_llm_config(cfg.model_dump())
    return {"status": "ok", "model": cfg.model, "base_url": cfg.base_url}
